lnwallet/wallet.go

```diff
@@ -1765,6 +1870,88 @@ func (l *LightningWallet) verifyFundingInputs(fundingTx *wire.MsgTx,
 	return nil
 }
 
+// verifyCommitSig verifies an incoming signature for our version of the
```

Comment:
could also do `verifyCommitSigV1/signCommitSigV1` and `verifyCommitSigLegacy/signCommitSigLegacy` since they are very different. This also made me think about whether it's worth adding dedicated tapscript methods to `CommitScriptToSelf` `CommitScriptToRemote` and `CommitScriptAnchors`, probably could make the overflow simpler?

Reply:
> whether it's worth adding dedicated tapscript methods to CommitScriptToSelf CommitScriptToRemote and CommitScriptAnchors, probably could make the overflow simpler?


You mean something like this: https://github.com/lightningnetwork/lnd/pull/7344#discussion_r1092017400 ?

Reply:
> could also do verifyCommitSigV1/signCommitSigV1 and verifyCommitSigLegacy/signCommitSigLegacy since they are very different

Opted to use the unified function as a fasacde here so we can maintain the same control flow. Seems like either need to branch at the call site (v1 vs legacy), or here within the sign method.